editor/plugins/control_editor_plugin.cpp

```diff
@@ -531,37 +641,42 @@ void ControlEditorToolbar::_set_anchors_preset(Control::LayoutPreset p_preset) {
 	undo_redo->commit_action();
 }
 
-void ControlEditorToolbar::_set_container_h_preset(Control::SizeFlags p_preset) {
-	List<Node *> selection = editor_selection->get_selected_node_list();
+void ControlAnchorsEditorToolbar::_button_toggle_picker_mode(bool p_pressed) {
+	pick_anchors_and_offsets = p_pressed;
 
-	undo_redo->create_action(TTR("Change Horizontal Size Flags"));
-	for (Node *E : selection) {
-		Control *control = Object::cast_to<Control>(E);
-		if (control) {
-			undo_redo->add_do_method(control, "set_h_size_flags", p_preset);
-			undo_redo->add_undo_method(control, "_edit_set_state", control->_edit_get_state());
-		}
+	if (pick_anchors_and_offsets) {
+		picker_mode_button->set_text(TTR("Anchors and offsets"));
+		picker_mode_button->set_tooltip(TTR("Disable to change only anchors."));
+	} else {
+		picker_mode_button->set_text(TTR("Anchors only"));
+		picker_mode_button->set_tooltip(TTR("Enable to change both anchors and offsets."));
 	}
-
-	undo_redo->commit_action();
 }
 
-void ControlEditorToolbar::_set_container_v_preset(Control::SizeFlags p_preset) {
-	List<Node *> selection = editor_selection->get_selected_node_list();
+void ControlAnchorsEditorToolbar::_picker_preset_selected(int p_preset) {
+	Control::LayoutPreset preset = (Control::LayoutPreset)p_preset;
+	if (pick_anchors_and_offsets) {
+		_set_anchors_and_offsets_preset(preset);
+	} else {
+		_set_anchors_preset(preset);
+	}
+}
 
-	undo_redo->create_action(TTR("Change Horizontal Size Flags"));
-	for (Node *E : selection) {
-		Control *control = Object::cast_to<Control>(E);
-		if (control) {
-			undo_redo->add_do_method(control, "set_v_size_flags", p_preset);
-			undo_redo->add_undo_method(control, "_edit_set_state", control->_edit_get_state());
+void ControlAnchorsEditorToolbar::_button_toggle_anchor_mode(bool p_status) {
+	List<Control *> selection = _get_edited_controls();
+	for (Control *E : selection) {
+		if (Object::cast_to<Container>(E->get_parent())) {
+			continue;
 		}
+
+		E->set_meta("_edit_use_anchors_", p_status);
```

Comment:
Unless the meta is necessary here, it could be removed on `false`. See #57863

Reply:
I just moved the method in this PR, so let's fix it in #57863 and then I'll rebase.